pkg/asset/installconfig/aws/validation.go

```diff
@@ -329,12 +330,22 @@ func validateEndpointAccessibility(endpointURL string) error {
 	if port == "" {
 		port = "https"
 	}
-	conn, err := proxy.Dial(context.Background(), "tcp", net.JoinHostPort(URL.Hostname(), port))
-	if err != nil {
-		return err
+	err = dialURL(context.Background(), net.JoinHostPort(URL.Hostname(), port))
+	return err
+}
+
+func dialURL(ctx context.Context, URL string) error {
+	// DefaultClient picks the http and https proxy environment variables.
+	_, err := http.Get(URL)
```

Comment:
I suggest HEAD rather than GET, since we do not care about any content that may be returned.